database/src/ops/blockchain.rs

```diff
@@ -1,9 +1,41 @@
 //! Blockchain.
 
 //---------------------------------------------------------------------------------------------------- Import
+use monero_serai::transaction::Timelock;
+
+use cuprate_types::VerifiedBlockInformation;
+
+use crate::{
+    database::{DatabaseRo, DatabaseRw},
+    env::EnvInner,
+    error::RuntimeError,
+    ops::macros::doc_error,
+    tables::{
+        BlockBlobs, BlockHeights, BlockInfoV1s, BlockInfoV2s, BlockInfoV3s, KeyImages, NumOutputs,
+        Outputs, PrunableHashes, PrunableTxBlobs, PrunedTxBlobs, RctOutputs, Tables, TablesMut,
+        TxHeights, TxIds, TxUnlockTime,
+    },
+    transaction::{TxRo, TxRw},
+    types::{
+        BlockHash, BlockHeight, BlockInfoLatest, BlockInfoV1, BlockInfoV2, BlockInfoV3, KeyImage,
+        Output, PreRctOutputId, RctOutput,
+    },
+};
 
 //---------------------------------------------------------------------------------------------------- Free Functions
-/// TODO
-pub fn height() {
-    todo!()
+/// Retrieve the block height of the latest/top block in the database.
+///
+/// # Example
+/// ```rust
+/// # use cuprate_database::{*, tables::*, ops::block::*};
+/// // TODO
+/// ```
+#[doc = doc_error!()]
+#[inline]
+pub fn height(
+    table_block_heights: &impl DatabaseRo<BlockHeights>,
+) -> Result<BlockHeight, RuntimeError> {
+    // TODO: is this correct?
+    // TODO: is there a faster way to do this? `.len()` is already quite cheap.
+    table_block_heights.len()
 }
```

Comment:
Correct?

Reply:
yeah but this should be called `chain_height` to make it clear that this returns the height of the chain and not the height of the top block